### ISSUE TITLE
About surface energy

### ISSUE THREAD

**Issue by casstudent** [shankar1729/jdftx#31]
Hi shankar，
       I have some trobules during calculating surface energy.(This may sound like a stupid question , but it really bother me!)
       After Test compiled executable I got a file: /jdftx/build/test/metalSurface/netural.out,
[neutral.out.txt](https://github.com/shankar1729/jdftx/files/2180147/neutral.out.txt)
and I tried to calculate Esurf from this output file :
△E = ( Etot(netural.out))-3*-89.035087)= 0.076361 Hartree = 2.0778 eV  
A= unit cell volume* 2/Lz= 475.518* 2/20 =47.5118 bohr^2 = 47.5118*0.05292^2 nm2 = 0.13317  nm2
Esurf =△E/A= 2.0778/ 0.13317 = 15.602 eV/nm2  （totally different from Tutorial：Metal surfaces ， in which the Esurf of Pt（111）is around 5 eV/nm2）
Could you point out my mistake?

Thanks for helping!

**Comment by shankar1729** [comment]
Calculate the bulk energy using the same pseudopotential and functional. You appear to have used PBEsol for surface and PBE for bulk.

**Comment by casstudent** [comment]
Hi shankar,
       I calculate the bulk energy using PBEsol and the result is Ebulk = -89.0304 Hartree (1 Pt atom)
[pt-bulk-12.out.txt](https://github.com/shankar1729/jdftx/files/2182615/pt-bulk-12.out.txt)
      Let it replace 89.035087 above, the final result will be  12.7 eV/nm2 
      Still far from the ideal result

**Comment by shankar1729** [comment]
Sorry, I didn't notice you were comparing the test output and I forgot that I had already changed all the metal tutorials to pbesol :).

The test calculations are all designed to run within a minute or so. Therefore they have smaller unit cells, fewer k points etc. Their goal is only to make sure the code gives the same answer on all machines and that we don't inadvertently change the answers when we modify the code. 

In particular, this one has only 3 layers and 8*8*1 k points, compared to 5 and 12*12*1 in tutorial. You can vary each of these and check which is responsible for most of the difference.

Best,
Shankar

**Comment by casstudent** [comment]
Hi shankar,
     Thank you for your prompt reply! 
     This time I used the input file in tutorial-metal surface , which contains 5 layers (1 atom per layer) and the kpoint is 12 12 1.  I didn't change anything , just copy and paste. The output file is 
[pt111-Vacuum-tutorial.out.txt](https://github.com/shankar1729/jdftx/files/2187597/pt111-Vacuum-tutorial.out.txt)
The total energy of pt111 surface is Etot = -445.093 Hartree  Ebulk = -89.0304 Hartree
△E =  Etot-5 * Ebulk= 0.059 Hartree = 1.605 eV
A= unit cell volume* 2/Lz= 855.932* 2/36 =47.5518 bohr^2 = 47.5518*0.05292^2 nm2 = 0.13317 nm2
Esurf =△E/A= 1.605/ 0.13317 = 12.05 eV/nm2 

      Still not right!  Something goes wrong but I can't figure out.
      Could you show me your output file , which get the right answer(5 ev/nm2). So that I will find the mistakes in my calculation.
      Thanks!

**Comment by shankar1729** [comment]
You should be picking up F (the Helmholtz free energy) not Etot from your output files, because this is a metal with smearing.

Best,
Shankar

**Comment by casstudent** [comment]
Hi Shankar，
    The result using F instead of Etot is 11.26eV/nm2， also far from the ideal result.

**Comment by shankar1729** [comment]
Indeed, it is! I found an inconsistency in my files used to provide the sample numbers in the tutorial. I somehow had Cold smearing in the bulk calculation whose energy went into the collectResults script. After fixing the bulk energy in the script, the surface energies line up with what you report above. (This was particularly confusing at the end there, because the F and Etot are very close with Cold smearing.)

Anyway, thanks a bunch for catching this error! I have updated the numbers on the tutorial.

Best,
Shankar
